### PR TITLE
[WIN32K:NTGDI] REGION_bXformRgn(): Restore check of last rectangle. CORE-15992

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

"Do not fix no regression"...
See #1798.

Addendum to c1ec0d92e24c9a27eb5ed2eae78268dda06626ad.

JIRA issue: [CORE-15992](https://jira.reactos.org/browse/CORE-15992)

### CODE REVIEW

**Comment by jimtabor** [comment]
Will this fix CORE-16299 ?

**Comment by SergeGautherie** [comment]
> Will this fix CORE-16299 ?

I see no relation to that issue.
Anyway, this PR shouldn't "fix" anything: it adds one safety check only.

File: win32ss/gdi/ntgdi/region.c [hunk 1]
```diff
@@ -2135,6 +2135,7 @@ REGION_bXformRgn(
             {
                 /* Make sure the rect is well-ordered after the xform */
                 RECTL_vMakeWellOrdered(&prgn->Buffer[i]);
+                NT_ASSERT(prgn->Buffer[i].top <= prgn->Buffer[i].bottom);
```
Comment (tkreuzer) [inline win32ss/gdi/ntgdi/region.c]: This assert is pointless, because that is what the above function will do.

File: win32ss/gdi/ntgdi/region.c [hunk 2]
```diff
@@ -2149,7 +2150,6 @@ REGION_bXformRgn(
             /* Loop all rects in the region */
             for (i = 0; i < prgn->rdh.nCount - 1; i++)
             {
-                NT_ASSERT(prgn->Buffer[i].top <= prgn->Buffer[i].bottom);
```
Comment (tkreuzer) [inline win32ss/gdi/ntgdi/region.c]: This assert is there for a reason!

**Comment by tkreuzer** [comment]
Useless assert added, useful assert removed. This is wrong.

File: win32ss/gdi/ntgdi/region.c (hunk 1 above)
Comment (SergeGautherie) [reply win32ss/gdi/ntgdi/region.c]: I concur with that.

File: win32ss/gdi/ntgdi/region.c (hunk 2 above)
Comment (SergeGautherie) [reply win32ss/gdi/ntgdi/region.c]: Well, that reason is not obvious to me: again, why is it correct not to check the last rectangle (anymore)?

**Pull request closed without merging**